### PR TITLE
Generate secrets with unique name

### PR DESCRIPTION
**Pull request by josecordaz**
**What changed?**
We are creating git secrets per repository.

**Why?**
We were overwriting git secrets causing authentication problems when flux tries to clone the repository.

**How did you test it?**
This was tested manually.

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: pkg/services/app/add.go [hunk 1]
```diff
@@ -317,36 +323,36 @@ func (a *App) commitAndPush(params AddParams, filters ...func(string) bool) erro
 	return nil
 }
 
-func (a *App) createAndUploadDeployKey(reposUrls []string, clusterName string, namespace string, dryRun bool) (string, error) {
-	secretRef := fmt.Sprintf("weave-gitops-%s", clusterName)
-	if dryRun {
-		return secretRef, nil
+func (a *App) createAndUploadDeployKey(repoUrl string, clusterName string, namespace string, dryRun bool) (string, error) {
+
+	if repoUrl == "" {
```
Comment (jrryjcksn) [inline pkg/services/app/add.go:328]: Is this a valid case? Should we return an error if `repoUrl` is empty?

**Review by jrryjcksn — APPROVED** [review]
One small question, but LGTM

File: pkg/services/app/add.go (hunk 1 above)
Comment (josecordaz) [reply pkg/services/app/add.go:328]: I'm with you, I think there is a proper way to do this. I was trying to fix a failure in the unit tests. But that would involve a bit more changes. I prefer to have this merged first as I need to talk with @luizbafilho to clarify some doubts.

Comment (luizbafilho) [reply pkg/services/app/add.go:328]: not a valid case, that's left over the UX change, feel free to remove it.

**Pull request merged**